src/Microsoft.IdentityModel.Tokens/ECDsaSecurityKey.cs

```diff
@@ -42,30 +42,20 @@ public ECDsaSecurityKey(byte[] blob, CngKeyBlobFormat blobFormat)
 
             CngKey = CngKey.Import(blob, blobFormat);
             BlobFormat = blobFormat;
+            HasPrivateKey = BlobFormat.Format == CngKeyBlobFormat.EccPrivateBlob.Format || BlobFormat.Format == CngKeyBlobFormat.GenericPrivateBlob.Format;
```

Comment:
I don't think keeping this CNG (i.e Windows)-specific constructor is the right thing to do. The public contract should only take OS/platform-agnostic parameters (i.e an `ECDsa` instance).